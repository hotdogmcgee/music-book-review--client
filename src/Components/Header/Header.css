.App__header {
  margin-bottom: 1rem;
  font-family: Laila;
}

.Header__nav {
  height: 40px;
  display: flex;
  align-items: normal;
  justify-content: space-between;
  padding: 10px;
  margin-bottom: 2rem;
}

.Header__nav h1 {
  margin: 0;
  font-size: 16px;
}

.Header__nav a {
  color: #444;
  text-decoration: none !important;
}

.Header__nav a:hover {
  color: none !important;
}

.Header__tagline--wide {
  opacity: 0.35;
  margin-left: 10px;
}

.Header__tagline--narrow {
  opacity: 0.35;
  display: block;
  text-align: center;
  display: none;
  margin-top: 10px;
}

.Header__not-logged-in {
  display: flex;

  align-items: center;
  justify-content: flex-end;
}

.Header__not-logged-in h2 {
  cursor: pointer;
}

.Header__not-logged-in a:last-child {
  background: rgb(255, 121, 121);
  color: white;
  padding: 5px 15px;
  border-radius: 15px;
  margin-left: 10px;
}

.Header__logged-in a {
  border: 1px solid rgba(0, 0, 0, 0.25);
  padding: 5px 15px;
  border-radius: 15px;
}

.Header__logged-in a:first-child {
  margin-right: 3px;
}

.login-container {
  display: flex;
  text-align: end;
  cursor: pointer;
}

.browse-and-search-container {
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 70px;
}

.show-dropdown {
  box-shadow: -7px 8px 15px -3px rgba(0, 0, 0, 0.61);
}

.hide-dropdown {
  display: none;
}

.Header__tagline--narrow {
  display: inherit;
}

.Header__tagline--wide {
  display: none;
}

.menu-wide {
  display: none;
}

@media screen and (min-width: 450px) {
  .Header {
    align-items: baseline;
  }

  .Header h1 {
    font-size: 22px;
  }
}

@media screen and (max-width: 400px) {
  .Header__logged-in a {
    padding: 5px 10px;
    font-size: 14px;
  }
}

@media screen and (min-width: 750px) {
  .Header__tagline--narrow {
    display: none;
  }
  .Header__tagline--wide {
    display: block;
    text-align: center;
    font-size: 16px;
    margin-left: auto;
  }

  .Header__nav h1 {
    font-size: 24px;
  }
}

@media screen and (min-width: 850px) {
  .App__header {
    margin-bottom: 100px;
  }

  .browse-and-search-container {
    position: absolute;
    top: 0;
    flex-direction: row-reverse;
    width: 60%;
    right: 0;
    padding: 0 10px;
  }

  .dropdown {
    height: 60px;
    width: 200px;
    display: flex;
    align-items: center;
  }

  .menu-wide {
    display: block;
    padding-top: 20px;
  }

  .menu-wide-list {
    list-style: none;
    display: flex;
    justify-content: flex-end;
    background: #373a47;
    padding: 20px;
  }

  .menu-wide-list-item {
    padding: 0 20px;
  }

  .menu-wide-list-item:hover {
    cursor: pointer;
  }
}

@media screen and (min-width: 1150px) {
  .browse-and-search-container {
    width: 56%
  }

  .dropdown > .button {
    font-size: 22px;
    height: 50px;
  }
}
