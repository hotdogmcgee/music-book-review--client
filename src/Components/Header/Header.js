import React from "react";
import { Link } from "react-router-dom";
import SearchBar from "./SearchBar/SearchBar";
import BrowseDropdown from "./BrowseDropdown/BrowseDropdown.js";
import TokenService from "../../services/token-service";
import BurgerMenu from "./BurgerMenu/BurgerMenu";
import BookListContext from "../../Contexts/BookListContext";

import "./Header.css";

export default class Header extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showLoginModal: false,
      hideElements: false
    };
    this.handleSearchBarFocus = this.handleSearchBarFocus.bind(this);
  }

  static defaultProps = {
    match: { params: {} }
  };
  static contextType = BookListContext;

  handleSearchBarFocus(bool) {
    this.setState({
      hideElements: !bool
    });
  }

  showModal = () => {
    this.setState({ showLoginModal: true });
  };

  hideModal = () => {
    this.setState({ showLoginModal: false });
  };
  handleLogoutClick = () => {
    TokenService.clearAuthToken();
    TokenService.clearUserId();
    this.props.hasLogin(false);
  };

  renderLogoutLink() {
    return (
      <div className="Header__logged-in">
        <Link onClick={this.handleLogoutClick} to="/">
          Logout
        </Link>
        <Link to="/my-profile">My Profile</Link>
      </div>
    );
  }

  renderLoginLink() {
    return (
      <div className="Header__not-logged-in" onClick={this.toggleBurger}>
        <div onClick={this.showModal}>
          <h2>
            <span>Log In /</span>
            <span>Register</span>
          </h2>
        </div>
      </div>
    );
  }

  renderBurgerMenu() {
    return <BurgerMenu showModal={this.showModal} />;
  }

  render() {
    return (
      <>
        <nav className="Header__nav">
          <h1>
            <Link to="/">Music Books Review</Link>
          </h1>
          <span className="Header__tagline--wide">
            For Students, Educators, and Parents!
          </span>
        </nav>
        <span className="Header__tagline--narrow">
          For Students, Educators, and Parents!
        </span>

        <div className="browse-and-search-container">
          <div className="search-bar-container">
            {" "}
            {this.props.showSearchBar ? (
              <SearchBar onSearchBarFocus={this.handleSearchBarFocus} />
            ) : (
              ""
            )}{" "}
          </div>

          <div className="show-dropdown">
            <BrowseDropdown />
          </div>
        </div>
      </>
    );
  }
}
