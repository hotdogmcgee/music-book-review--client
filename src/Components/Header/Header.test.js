import React from 'react';
import Header from './Header.js'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

describe('Header component', () => {

    it('renders a Header by default', () => {
        const wrapper = shallow(<Header />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it("renders the UI as expected", () => {
        const tree = renderer
          .create(
            <BrowserRouter>
              <Header />
            </BrowserRouter>
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

})