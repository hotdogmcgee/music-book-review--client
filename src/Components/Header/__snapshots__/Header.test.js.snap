// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders a Header by default 1`] = `
<Fragment>
  <nav
    className="Header__nav"
  >
    <div>
      <h1>
        <Link
          to="/"
        >
          Music Book Reviews
        </Link>
      </h1>
      <span
        className="Header__tagline--wide"
      >
        For Students, Educators, and Parents!
      </span>
    </div>
  </nav>
  <span
    className="Header__tagline--narrow"
  >
    For Students, Educators, and Parents!
  </span>
  <div
    className="browse-and-search-container"
  >
    <div
      className="search-bar-container"
    >
       
       
    </div>
    <div
      className="show-dropdown"
    >
      <BrowseDropdown
        history={
          Object {
            "push": [Function],
          }
        }
      />
    </div>
  </div>
  <div
    className="menu-wide"
  >
    <ul
      className="menu-wide-list"
    >
      <li
        className="menu-wide-list-item"
      >
        <div
          className="menu-item"
          id="log-in"
          onClick={[Function]}
        >
          Log In/Register
        </div>
      </li>
      <li
        className="menu-wide-list-item"
      />
      <li
        className="menu-wide-list-item"
      >
        <Link
          className="menu-item"
          id="about"
          to="/about"
        >
          About
        </Link>
      </li>
    </ul>
  </div>
</Fragment>
`;

exports[`Header component renders the UI as expected 1`] = `
Array [
  <nav
    className="Header__nav"
  >
    <div>
      <h1>
        <a
          href="/"
          onClick={[Function]}
        >
          Music Book Reviews
        </a>
      </h1>
      <span
        className="Header__tagline--wide"
      >
        For Students, Educators, and Parents!
      </span>
    </div>
  </nav>,
  <span
    className="Header__tagline--narrow"
  >
    For Students, Educators, and Parents!
  </span>,
  <div
    className="browse-and-search-container"
  >
    <div
      className="search-bar-container"
    >
       
      
       
    </div>
    <div
      className="show-dropdown"
    >
      <div
        className="dropdown"
      >
        <div
          className="button"
          onClick={[Function]}
        >
          Browse
        </div>
      </div>
    </div>
  </div>,
  <div
    className="menu-wide"
  >
    <ul
      className="menu-wide-list"
    >
      <li
        className="menu-wide-list-item"
      >
        <div
          className="menu-item"
          id="log-in"
          onClick={[Function]}
        >
          Log In/Register
        </div>
      </li>
      <li
        className="menu-wide-list-item"
      >
        
      </li>
      <li
        className="menu-wide-list-item"
      >
        <a
          className="menu-item"
          href="/about"
          id="about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
    </ul>
  </div>,
]
`;
